pkg/volume/csi/csi_attacher.go

```diff
@@ -197,8 +197,18 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 		}
 
 		attachID := getAttachmentName(volumeHandle, driverName, string(nodeName))
+		var attach *storage.VolumeAttachment
+		if c.plugin.volumeAttachmentLister != nil {
+			attach, err = c.plugin.volumeAttachmentLister.Get(attachID)
+			if err == nil {
+				attached[spec] = attach.Status.Attached
+				continue
+			}
```

Comment:
Should we log the error?

Reply:
Please handle NotFound error here - such volume is detached and we don't need logging (or we can use less serious logging level).